linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/IncOptimizer.scala

```diff
@@ -88,14 +88,18 @@ final class IncOptimizer private[optimizer] (config: CommonPhaseConfig, collOps:
       processAllTaggedMethods(logger)
     }
 
-    val newLinkedClasses = unit.classDefs.map(optimizedLinkedClass(_))
-    val newTopLevelExports = unit.topLevelExports.map(optimizedTopLevelExport(_))
+    val groupedTopLevelExports = unit.topLevelExports.groupBy(_.owningClass)
```

Comment:
Here we re-group top level exports just to un-group them right away later in refiner.

Reply:
Hum I see. Perhaps this could be improved (later) in the definition of `LinkingUnit`. The latter could associate top-level export defs with their respective `LinkedClass`es?

Reply:
I'll take a look at that option. However, we split top level exports off deliberately for module splitting: We need to be able to put them into a different module than their owning class.

Reply:
If it doesn't make things cleaner overall, we can also leave it as is. The group-ungroup is slightly inelegant but better than the optimizer "lie" that we currently have, IMO.

Reply:
Yeah, agreed. So let's go forward with this and I'll follow-up.